src/app/shared/components/template/components/audio/audio.component.html

```diff
@@ -5,9 +5,15 @@ <h3>{{ params.title }}</h3>
       @if (params.variant.includes("large") && params.showInfoButton) {
         <div class="info-icon-container">
           @if (params.infoIconAsset) {
-            <ion-icon [src]="params.infoIconAsset | plhAsset" (click)="clickInfo()"></ion-icon>
+            <ion-icon
+              [src]="params.infoIconAsset | plhAsset"
+              (click)="params.transcriptText ? openTranscriptPopup() : clickInfo()"
+            ></ion-icon>
           } @else {
-            <ion-icon name="document-text-outline" (click)="clickInfo()"></ion-icon>
+            <ion-icon
+              name="document-text-outline"
+              (click)="params.transcriptText ? openTranscriptPopup() : clickInfo()"
+            ></ion-icon>
```

Comment:
nit(non-blocking): I'm never a fan of attaching click events directly to icons as they often don't have a very noticeable (or accessible) clickable region. Would instead recommend using a `ion-button` element for the container and attaching the click-handler to it (also less duplication). At the same time it's probably better to just write a simple `handleActionButtonClick` method and deal with the conditional logic there, as you've got 4 inline handlers that will re-evaluate every render even when no button is clicked (not a huge deal, but should just help keep the HTML a little cleaner). E.g.

```html
<ion-button fill="clear" (click)="handleActionButtonClick()">
          @if (params.infoIconAsset) {
            <ion-icon slot="icon-only" [src]="params.infoIconAsset | plhAsset"></ion-icon>
          } @else {
            <ion-icon slot="icon-only" name="document-text-outline"></ion-icon>
          }
        </ion-button>
 ```
 You might also find that with the inherited ion-button styles you don't actually need the custom style class anymore
 
 If you didn't want to duplicate the div you could define it as a named `ng-template` and use `ng-container` to inject into both sections (https://angular.dev/guide/templates/ng-template#using-ngtemplateoutlet), although might be overkill

Reply:
Thanks, that's much neater now, see https://github.com/IDEMSInternational/open-app-builder/pull/2460/commits/5ecc580258108370b5f327586b41c0531fa134c2